src/kotlin_editor_export_plugin.cpp

```diff
@@ -9,97 +9,91 @@
-            files_to_add.push_back("res://build/libs/godot-bootstrap.jar");
-            files_to_add.push_back(vformat("res://build/libs/%s", graal_usercode_lib));
-            _generate_export_configuration_file(GDKotlin::get_instance().get_configuration().vm_type);
+    bool is_desktop_export {is_windows_export || is_linux_export || is_osx_export};
+    bool is_android_export {p_features.has("android")};
+    bool is_ios_export {p_features.has("ios")};
 
-        } else {
-            if (FileAccess::exists(configuration_path)) {
-                Ref<FileAccess> configuration_access_read {FileAccess::open(configuration_path, FileAccess::READ)};
-                jni::JvmType jvm_type {GDKotlin::get_instance().get_configuration().vm_type};
-                switch (jvm_type) {
-                    case jni::JvmType::JVM:
-                        files_to_add.push_back("res://build/libs/main.jar");
-                        files_to_add.push_back("res://build/libs/godot-bootstrap.jar");
-                        _generate_export_configuration_file(jni::JvmType::JVM);
-
-                        break;
-                    case jni::JvmType::GRAAL_NATIVE_IMAGE:
-                        files_to_add.push_back(vformat("res://build/libs/%s", graal_usercode_lib));
-                        _generate_export_configuration_file(jni::JvmType::GRAAL_NATIVE_IMAGE);
-                        is_graal_only = true;
-
-                        break;
-                    default:
-                        LOG_ERROR("Incorrect VM type for desktop, aborting export.");
-                        return;
-                }
-            } else {
-                LOG_ERROR(vformat("Cannot find configuration file %s", configuration_path));
-            }
-        }
-    }
+    if (is_desktop_export) {
+        bool export_all {p_features.has(all_jvm_feature)};
+        bool export_graal {p_features.has(graal_feature) || export_all};
+        bool export_jvm {!p_features.has(graal_feature) || export_all};
+
+        if (export_jvm) {
+            files_to_add.push_back(String(BUILD_DIRECTORY).path_join(DESKTOP_BOOTSTRAP_FILE));
+            files_to_add.push_back(String(BUILD_DIRECTORY).path_join(DESKTOP_USER_CODE_FILE));
 
-    // Copy JRE for desktop platforms
-    if (!is_android_export && !is_graal_only) {
-        const Vector<String>& path_split = p_path.split("/");
-        String export_dir {p_path.replace(path_split[path_split.size() - 1], "")};
-        Error error;
-        Ref<DirAccess> dir_access {DirAccess::open(export_dir, &error)};
-        if (error == OK) {
             if (is_osx_export) {
                 bool is_arm64 {p_features.has("arm64")};
                 bool is_x64 {p_features.has("x86_64")};
 
                 if (!is_arm64 && !is_x64) {
-                    add_macos_plugin_file(vformat("res://%s", EMBEDDED_JRE_DIRECTORY));
+                    add_macos_plugin_file(String(RES_DIRECTORY).path_join(EMBEDDED_JRE_DIRECTORY));
```

Comment:
more of a question: will this not result in: `res:///path` (note the third /)?

Reply:
`path_join` is a function explicitly made to handle such case.
https://docs.godotengine.org/fr/4.x/classes/class_string.html#class-string-method-path-join